### PR TITLE
New Visitor class methods for preferred MIME type

### PR DESCRIPTION
**Pull request by lukasbestle**
This is the first set of changes from #2148. Removing the deprecated features requires these changes, so I will send the other two PRs once this one is merged.

### CODE REVIEW

**Comment by distantnative** [comment]
I think it might be fine for this PR, but the removal of lines like

```php
$this->assertTrue($page->hasVisibleChildren());
```

should go into it's own PR to not distract while reading the changes and new implementations.

Also the name of the PR seems off since the main part is around preferred MIME type.

**Comment by lukasbestle** [comment]
@distantnative It's very difficult to structure those PRs properly. The more we split it up, the longer it takes because I can only create the next PR once the previous one has been merged. Removing the tests for example is required for the next PR that removes the deprecated methods, otherwise the tests won't pass.

Regarding the naming: The preferred MIME type feature has only been implemented so that we can migrate from the old `$request->ajax()` method that has been deprecated. So same thing here: If I'd split that up, that would be another PR that would need to be merged before this change could be made. It's already three PRs (two of them being lined up) and I really don't want it to be even more confusing for us all by splitting it up more than it needs to be.

**Comment by distantnative** [comment]
Not sure if I follow your steps correctly:
Why can't we bundle removing 

```php
$this->assertTrue($page->hasVisibleChildren());
```

with removing the actual deprecated methods in one PR? The tests should pass when you remove them in the same PR. And bundling would make it much more sensible to me.

I get that splitting PRs can be annoying for the one working on them - trust me @bastianallgeier has pushed me to it over and over cause I tend to make huge PRs XD But from the perspective of one reviewing a PR it's really confusing to see code changes that don't really belong to the rest of the PR.

**Comment by lukasbestle** [comment]
Oh, right! I actually meant to do that but it looks like I moved the changes to the wrong branch when splitting the PRs. Will fix that now.

File: src/Toolkit/Mime.php [hunk 1]
```diff
@@ -239,7 +240,20 @@ public static function isAccepted(string $mime, string $pattern): bool
...
+     * matches a given type
+     *
+     * @param string $wildcard
+     * @param string $test
+     * @return bool
+     */
+    public static function matchWildcard(string $wildcard, string $test): bool
+    {
+        return fnmatch($wildcard, $test, FNM_PATHNAME) === true;
+    }
```
Comment (bastianallgeier) [inline src/Toolkit/Mime.php]: I'd like to restructure this a bit. 

```php
public static function matches(string $mime, string $wildcard): bool
{
    return fnmatch($wildcard, $mime, FNM_PATHNAME) === true;
}
```

In my opinion, it fits better to the way we normally handle arguments in such static methods. i.e. see `Mime::isAccepted` or something like `Str::contains`. We always pass the "haystack" first and then the "needle" second. 

```
Mime::matches('text/html', 'text/*');
```

**Review by bastianallgeier — CHANGES_REQUESTED** [review]

File: src/Toolkit/Mime.php (hunk 1 above)
Comment (bastianallgeier) [reply src/Toolkit/Mime.php]: The rest is great!

**Review by bastianallgeier — APPROVED** [review]
I just made the changes to the Mime class. The rest is perfect!

**Pull request merged**